k8s/upgrade/src/upgrade_resources/upgrade.rs

```diff
@@ -97,6 +97,10 @@ pub struct UpgradeArgs {
     /// Upgrade to the specified branch.
     #[clap(global = true, hide = true, long, short = 'u')]
     pub upgrade_to_branch: Option<String>,
+
+    /// Upgrade tag for testing, contains timestamp.
+    #[clap(global = true, hide = true, long)]
+    pub upgrade_tag_for_test_envt: Option<String>,
```

Comment:
Is it a good idea to have a new variable for test environment inside production code? Is it normal?

Reply:
In the meeting it was discussed that we can introduce a flag , which this PR is doing. Since this flag is hidden, users cant use is unless they go through the code. The binary wont show this with help command.

Reply:
Do we plan to feed the image tag timestamp suffix to the upgrade command, or the complete image tag... ?

Reply:
It seems like this flag could only possibly work with the 'upgrade' subcommand, and not the 'delete upgrade' or 'get upgrade-status' subcommands. The flag is maybe not a global flag...?

Reply:
> Is it a good idea to have a new variable for test environment inside production code? Is it normal?
No, it's not usually a good idea... though not sure exactly here, may be no other way?

As an alternative, when we built these timestamped images, can't we just set the target to the timestampeped tag?

Reply:
modified the code . now the user can directly pass the tag , rather than appending the time stamp.